src/playback-watcher.js

```diff
@@ -356,17 +370,22 @@ export default class PlaybackWatcher {
       return true;
     }
 
+    const sourceUpdater = this.masterPlaylistController_.sourceUpdater_;
     const buffered = this.tech_.buffered();
 
     if (
       closeToBufferedContent({
         buffered,
+        audioBuffered: sourceUpdater.audioBuffer ? sourceUpdater.audioBuffered() : null,
+        videoBuffered: sourceUpdater.videoBuffer ? sourceUpdater.videoBuffered() : null,
         targetDuration: this.media().targetDuration,
         currentTime
       })
     ) {
-      seekTo = buffered.start(0) + Ranges.SAFE_TIME_DELTA;
-      this.logger_(`Buffered region starts (${buffered.start(0)}) ` +
+      const nextRange = Ranges.findNextRange(buffered, currentTime);
```

Comment:
Example values for why we need this:
```
buffered = [[0, 10], [21, 40]]
currentTime = 20
````
Previously we would just take `buffered.start(0)` which is in this case is before currentTime. Now we find the `TimeRange` that is after `currentTime` and use that.